src/lexer/regexFactory.ts

```diff
@@ -112,23 +112,33 @@ const buildQStringPatterns = () => {
 // 7. BigQuery """triple-quoted"""
 // 8. Hive and Spark variables: ${name}
 export const quotePatterns = {
-  '``': '(?:`[^`]*(?:$|`))+',
-  '[]': '(?:\\[[^\\]]*(?:$|\\]))(?:\\][^\\]]*(?:$|\\]))*',
-  '""': '(?:"[^"\\\\]*(?:\\\\.[^"\\\\]*)*(?:"|$))+',
-  "''": "(?:'[^'\\\\]*(?:\\\\.[^'\\\\]*)*(?:'|$))+",
-  '$$': '(?<tag>\\$\\w*\\$)[\\s\\S]*?(?:\\k<tag>|$)',
-  "'''..'''": "'''[^\\\\]*?(?:\\\\.[^\\\\]*?)*?(?:'''|$)",
-  '""".."""': '"""[^\\\\]*?(?:\\\\.[^\\\\]*?)*?(?:"""|$)',
-  '{}': '(?:\\{[^\\}]*(?:$|\\}))',
+  '``': '`(?:[^`]|``)*(?:`|$)',
+  '[]': String.raw`\[(?:[^\]]|\]\])*(?:\]|$)`,
+  '""': String.raw`(?:"{escape}*?(?:\\.[^"\\]*)*(?:"|$))+`,
+  "''": String.raw`(?:'{escape}*?(?:\\.[^'\\]*)*(?:'|$))+`,
```

Comment:
I'm pretty confident this doesn't work as intended.

Let me simplify this regex a bit with additional whitespace and removal of  `:?` optional capture groups:

```
( '{escape}*? (\\.[^'\\]*)* ('|$) )+
```

a) When escapes array contains a single quote `'`, the resulting regex will be:

```
( '[^']*? (\\.[^'\\]*)* ('|$) )+
```

This regex will happily match the following string:

```
'hello\'world'
```

b) When escapes array contains a backslash, the resulting regex will be:

```
( '[^\\]*? (\\.[^'\\]*)* ('|$) )+
```

This regex happily matches the following string:

```
'bla'bla''bla'
```